src/models/events.model.ts

```diff
@@ -0,0 +1,13 @@
+
+// The events declared in the ServerToClientEvents interface are used when sending and broadcasting events
+export interface ServerToClientEvents {
+    noArg: () => void;
+    basicEmit: (a: number, b: string, c: Buffer) => void;
+    withAck: (d: string, callback: (e: number) => void) => void;
+}
+
+// The events declared in the ClientToServerEvents interface are used when receiving events
+export interface ClientToServerEvents {
```

Comment:
Interesting. So the socket server is going to use events for communication.

Is the actual transmission of events handled by socket.io? Or do we need to come up with a way of managing connections?

Reply:
We need to come up with a way of managing connections. Socket.io is really abstract in a way it will be scalable for developers.